### DISCUSSION TITLE
Devices/Camera improvements.

### DISCUSSION THREAD

**Discussion by skadlec72** [OrcaSlicer/OrcaSlicer#2998] [Ideas]
Ability to  resize the camera window and/or add the ability to pop the camera out into a separate window

**Comment by funkymonkeymonk** [comment]
I wanted to come support this and also provide an example situation where this is really relevant. I have a vertical monitor where I would love to put my slicer and also monitor my print. Unfortunately the device window looks like this.

![Screenshot 2024-04-22 at 8 07 26 PM](https://github.com/SoftFever/OrcaSlicer/assets/5369497/04bfdac1-414a-46bd-ae2c-729a71764203)

Because the Camera doesn't fit to size very well and the controls/print progress don't reflow I end up with a super useless view/control station. This means I have to put the slicer in the background on my main monitor and not view my progress. I'd love to see this improvement taken up or maybe an alternative workflow suggested.

**Comment by foresterr** [comment]
Isn't the "Device" tab just a web view to whatever interface your printer provides? E.g. in my case it's Mainsail, and if I wanted such a feature, it would have to be in Mainsail, not in Orca.

**Comment by rapapar** [comment]
> Ability to resize the camera window and/or add the ability to pop the camera out into a separate window

Same idea as I have a second monitor but smaller so I can not watch well on that small windows, it should be better in full screen and maybe if we can locally reach the camera (with an icmp ping) to use only lan and not external server (to get better fps)
